### ISSUE TITLE
Add overlays

### ISSUE THREAD

**Issue by zbraniecki** [unclenachoduh/ex_cldr#4]
When handling CLDR data, we want to take all the CLDR data and then apply one of the two kinds of overlays:

 - add locales that CLDR doesn't have
 - add overlays for some bit for locales we do.

I imagine us having an optional directory `./overlays/languages/*.ftl` and `./overlays/regions/*.ftl` where each locale may have its own FTL file with overlays.

For example, a file `./overlays/languages/it.ftl` could contain:

```properties
language-name-fr = Overlay Name For French
```

while for a locale that CLDR does not have, say `kl`, it may have `./overlays/languages/kl.ftl` with all the data:

```properties
language-name-az = ...
language-name-bc = ...
language-name-cd = ...
```

The script should take those overlays and apply them on top of the data it extracted from CLDR either replacing or adding the new bits.

The result directory should contain both CLDR and overlay locales.

**Comment by unclenachoduh** [comment]
Does the `overlays` folder belong in root or in the same folder as the script. Currently, there is no structure, so it's possible you meant both or neither.

**Comment by zbraniecki** [comment]
yeah, I think we can start with the overlay document being optionally expect to be in the same directory as the script. We can later extend it to allow for some argument to be passed to point to that dir in other locations.

**Comment by unclenachoduh** [comment]
@zbraniecki first implementation of overlays is complete with that last commit

**Comment by unclenachoduh** [comment]
The script doesn't apply overlay data "on top" of the CLDR data, necessarily. If a variable name from the overlay data (e.g. 'language-name-en') already exists in the CLDR data, the value from CLDR is replaced with the overlay value. All overlay variables that do not already exist in the CLDR data are appended to the end of the `resources.ftl` file.

**Comment by flodolo** [comment]
Commenting here, since the issue is still open. I think the structure of the *overlay* folder should be flipped: there should be a folder for each locale, containing files as needed.
* overlay/languages/ko.ftl -> overlay/ko/languages.ftl
* overlay/regions/ko.ftl -> overlay/ko/regions.ftl

That way, we could even use an external repository directly used by Pontoon for translations.

**Comment by flodolo** [comment]
Trying to keep comments on the coding style to a minimum (I guess the goal is to have a MVP, then clean it up). Also, I'm sure Zibi and I have quite different opinions, and what counts is Zibi's :)

Having said that, you should really parse the FTL file via Fluent, not splitting the lines in the Overlay.

Talking about overlays, I think it's still missing an important bit:

> while for a locale that CLDR does not have, say kl, it may have ./overlays/languages/kl.ftl with all the data

If a locale is completely missing on CLDR, we still need to be able to read our local overlays. I would say, we should even merge missing content from English if a string is missing, because I assume ICU wants complete files.

**Comment by flodolo** [comment]
One more thought: what if we want to overlay data even on top of English? For example, language names that are still missing from the current release of CLDR. Would that be in scope?

**Comment by unclenachoduh** [comment]
@zbraniecki: @flodolo 's restructure of the `overlays` folder is consistent with the current output folder. Should we move forward with that?

**Comment by unclenachoduh** [comment]
@flodolo: I think both overlay features you are requesting are already implemented in the script.

The current script does:
- create a new folder and .flt file for a language that exists in the `overlays` folder but not in the CLDR data.
- overwrite variables in the CLDR data with values from matching variables in the overlay data.

Does that match what you were describing?

**Comment by flodolo** [comment]
> Does that match what you were describing?

Partially, it's missing the last part: what if the overlay only contains 4 strings instead of 1k? Not sure if we want the FTL file to be complete, injecting English content, or that's something we would do at export time. 

It would seem safer to do at this point, but that's more a design choice for @zbraniecki
